### PR TITLE
Update example to use modern rust-gpu

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/EmbarkStudios/rust-gpu/blob/XAMPPRocky-patch-1/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,21 +19,39 @@ However, many things aren't implemented yet: for example, loops and switches are
...
-    
+use spirv_std::{glam::{Vec3, Vec4, vec2, vec3}, storage_class::{Input, Output}};
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+#[spirv(block)]
+pub struct ShaderConstants {
+    pub width: u32,
+    pub height: u32,
+}
```
Comment (Jasper-Bekkers) [inline README.md:30]: Not sure if we should include this in the readme, we removed it last time to keep the sample small.

File: README.md [hunk 2]
```diff
@@ -19,21 +19,39 @@ However, many things aren't implemented yet: for example, loops and switches are
...
+
+#[allow(unused_attributes)]
```
Comment (Jasper-Bekkers) [inline README.md:32]: We shouldn't need this anymore since 6571f45a

File: README.md [hunk 3]
```diff
@@ -19,21 +19,39 @@ However, many things aren't implemented yet: for example, loops and switches are
...
+    let eye_pos = vec3(0.0, 0.0997, 0.2);
+    let sun_pos = vec3(0.0, 75.0, -1000.0);
+    let dir = get_ray_dir(uv, eye_pos, sun_pos);
+
     // evaluate Preetham sky model
     let color = sky(dir, sun_pos);
 
-    output.store(color.extend(0.0))
+    // Tonemapping
+    let color = color.max(Vec3::splat(0.0)).min(Vec3::splat(1024.0));
```
Comment (Jasper-Bekkers) [inline README.md:52]: Can we use `clamp` here and/or remove this clamp, it seems incorrect?

**Review by Jasper-Bekkers — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (XAMPPRocky) [reply README.md]: I've removed it for now, but this is copy-pasted from the `sky-shader`.

**Review by Jasper-Bekkers — APPROVED** [review]

**Pull request merged**